x/amm/keeper/pool.go

```diff
@@ -183,6 +183,14 @@ func (k Keeper) GetPoolSnapshotOrSet(ctx sdk.Context, pool types.Pool) (val type
 	}
 
 	k.cdc.MustUnmarshal(b, &val)
+
+	// Update the pool snapshot with accounted pool balance
+	for i, asset := range val.PoolAssets {
```

Comment:
I am not sure if its good idea to set accounted pool balances here as its getter and setter function from KV store. Ideally should be done outside.
Even if we are doing it, at least function should specify this change.

Reply:
okay, let's do that, but this changes is required otherwise amm logic will be incorrect

Reply:
I will rename the function, but this is needed to avoid any confusion
NOte: `we should use accounted pool data at any place if it is available`, right ?

Reply:
added a new function with comments